src/bootstrap/src/core/build_steps/test.rs

```diff
@@ -680,10 +680,10 @@ impl Step for Miri {
             .arg("--manifest-path")
             .arg(builder.src.join("src/tools/miri/test-cargo-miri/Cargo.toml"));
         cargo.arg("--target").arg(target.rustc_target_arg());
-        cargo.arg("--tests"); // don't run doctests, they are too confused by the staging
-        cargo.arg("--").args(builder.config.test_args());
 
+        cargo.arg("--").args(builder.config.test_args());
         // Tell `cargo miri` where to find things.
+        cargo.env("RUSTDOC_REAL", builder.rustdoc(compiler_std));
```

Comment:
```suggestion
        // `prepare_tool_cargo` sets RUSTDOC to the bootstrap wrapper, but it did not set
        // RUSTDOC_REAL for us as this is a "run", not a "test". And anyway we want the rustdoc from
        // the "next" stage for the same reason that we build a std from the next stage. So let's
        // just set that here.
        cargo.env("RUSTDOC_REAL", builder.rustdoc(compiler_std));
        // Tell `cargo miri` where to find things.
```

Reply:
Had to do some updates on this due to other changes (now we are passing RUSTDOC not RUSTDOC_REAL)